src/lib/components/DeleteButton.js

```diff
@@ -113,19 +113,19 @@ export class DeleteButtonComponent extends Component {
             <Button onClick={this.closeConfirmModal} floated="left">
               {i18next.t('Cancel')}
             </Button>
-            <ActionButton {...color} name="delete" onClick={this.handleDelete}>
-              {(formik) => (
-                <>
-                  {formik.isSubmitting &&
-                  actionState === DRAFT_DELETE_STARTED ? (
-                    <Icon size="large" loading name="spinner" />
-                  ) : (
-                    <Icon name="trash alternate outline" />
-                  )}
-                  {capitalizedActionLbl}
-                </>
-              )}
-            </ActionButton>
+            <ActionButton
+              {...color}
+              name="delete"
+              onClick={this.handleDelete}
+              icon={(formik) =>
+                formik.isSubmitting && actionState === DRAFT_DELETE_STARTED ? (
+                  <Icon size="large" loading name="spinner" />
```

Comment:
loading should be a prop of a button not the icon... in this case I think you only need to specify when the button should be loading not to introduce the new loading icon